tests/test_localedata.py

```diff
@@ -73,6 +75,23 @@ def test_merge():
     localedata.merge(d, {1: 'Foo', 2: 'Bar'})
     assert d == {1: 'Foo', 2: 'Bar', 3: 'baz'}
 
+
 def test_locale_identification():
     for l in localedata.locale_identifiers():
         assert localedata.exists(l)
+
+
+def test_unique_ids():
+    # Check all locale IDs are uniques.
+    all_ids = localedata.locale_identifiers()
+    assert len(all_ids) == len(set(all_ids))
+    # Check locale IDs don't collide after lower-case normalization.
+    lower_case_ids = list(map(methodcaller('lower'), all_ids))
+    assert len(lower_case_ids) == len(set(lower_case_ids))
+
+
+def test_mixedcased_locale():
+    for l in localedata.locale_identifiers():
```

Comment:
That's another optimization that could be done (via `pytest.parametrize`), but I don't think our test suite takes too long to run as it is :)